web3/gas_strategies/time_based.py

```diff
@@ -20,7 +20,8 @@ def _get_avg_block_time(w3, sample_size):
     latest = w3.eth.getBlock('latest')
 
     if latest['number'] == 0 or sample_size == 0:
-        return 0
+        # a safe default block time
+        return 15
```

Comment:
@carver what do you think about allowing this to be configured via environment variable?

Reply:
cc @dylanjw too

Reply:
I think we should raise if you try to construct a 0 sample size middleware: https://github.com/glaksmono/web3.py/blob/fb464356a644445559d798258e585f13aa16964f/web3/gas_strategies/time_based.py#L123

Otherwise, estimating with 0 blocks available seems too trivial to worry about otherwise. I can't imagine anyone bothering to set an env variable for what to do when estimating gas for only the first block.